frontend/src/components/matching/MatchingLobby.tsx

```diff
@@ -44,143 +39,175 @@ export default function MatchingLobby({
     topics: string[],
   }
 }) {
-  const [stage, setStage] = React.useState(MATCHING_STAGE.INITIAL);
-  const { onOpenChange } = useDisclosure();
-
-  const debugStage = () => {
-    let opts = Object.keys(MATCHING_STAGE);
-    let next = opts.findIndex(x => x === stage.toString()) + 1 % opts.length;
-    setStage(next)
+  const initialSuccessState: MatchingSuccessState = {
+    userReady: false,
+    partnerReady: false,
+    partnerLeft: false,
+    owner: false,
+    partner: {
+      id: "",
+      name: "",
+      image: undefined
+    }
   }
 
-  // Trigger matching process on modal open
-  React.useEffect(() => {
-    if (isOpen) {
-      setStage(MATCHING_STAGE.MATCHING)
-    } else {
-      setStage(MATCHING_STAGE.INITIAL)
-    }
-  }, [isOpen])
+  const [stage, setStage] = React.useState(MATCHING_STAGE.INITIAL);
+  const [successState, setSuccessState] = React.useState(initialSuccessState);
 
-  // Response to stage events
-  React.useEffect(() => {
-    switch (stage) {
-      case MATCHING_STAGE.MATCHING:
-        onMatchingStage();
-        break;
-      default:
-        break;
-    }
-  }, [stage])
+  const [socket, setSocket] = React.useState<Socket>();
+  const { user } = useAuthContext();
 
-  // Handle view switching
-  const renderView = (stage: MATCHING_STAGE) => {
-    switch (stage) {
-      case MATCHING_STAGE.MATCHING:
-        return initialView;
-      case MATCHING_STAGE.SUCCESS:
-        return successView;
-      case MATCHING_STAGE.FAIL:
-        return failureView;
-      default:
-        return errorView;
+  const logger = getLogger('matching');
+
+  /////////////////////////////////////////////
+  // Stage fired events
+  /////////////////////////////////////////////
+  const initializeSocket = async () => {
```

Comment:
Can potentially move this to `helpers` too if this does not cause any breakage. By the way, for the socket events like `connect`, `disconnect`, `connect_error`, and etc., do you consider create an enum for them?